contrib/generic_proxy/filters/network/source/proxy.cc

```diff
@@ -382,8 +483,28 @@ Envoy::Network::FilterStatus Filter::onData(Envoy::Buffer::Instance& data, bool)
   return Envoy::Network::FilterStatus::StopIteration;
 }
 
-void Filter::onDecodingSuccess(RequestPtr request, ExtendedOptions options) {
-  newDownstreamRequest(std::move(request), options);
+void Filter::onDecodingSuccess(StreamFramePtr request) {
+  const uint64_t stream_id = request->frameFlags().streamFlags().streamId();
+  // One existing stream expects this frame.
+  if (auto iter = frame_handlers_.find(stream_id); iter != frame_handlers_.end()) {
+    iter->second->onRequestFrame(std::move(request));
+    return;
+  }
+
+  StreamFramePtrHelper<StreamRequest> helper(std::move(request));
+
+  // Create a new active stream for the leading StreamRequest frame.
+  if (helper.typed_frame_ != nullptr) {
+    newDownstreamRequest(std::move(helper.typed_frame_));
```

Comment:
This looks like we assume the request only use one frame, what happened if the request is composed by multiple frames?

Reply:
I think the code above your comment have handle the multiple frame condition.

If a new frame is received, we will check the frame_handlers_ first to see if there is a exist stream expect this frame. If there are no exist stream want it and it's StreamRequest frame, we will create a stream for it.